### DISCUSSION TITLE
lowest latency grab frame into opencv, & set exposure max limits

### DISCUSSION THREAD

**Discussion by LiellPlane** [raspberrypi/picamera2#543] [General]
Hi, thanks for the amazing library

What is the lowest latency method to grab an image and push it into opencv? Currently I am getting 90ms on a raspberry pi 4 with capture_array for a 2000*1500 image which is not fast enough for my real-time application. I do not need colour if that helps, just intensity

Bonus question - can I set a max exposure time, so the hardware can adjust to the light level until it hits a ceiling? Apologies if I have missed this in the documentation

**Comment by davidplowman** [comment]
Hi, the speed at which you can do this will depend on which sensor you are using, the mode chosen for the sensor, the exposure time, whether the sensor is already running, and so on. Can you provide any more information? A minimal test script so that we can see exactly what you're doing is often helpful too. Thanks!
